### ISSUE TITLE
Decoding speech from non-invasive brain recordings

### ISSUE THREAD

**Issue by Serendipityzzz** [Serendipityzzz/NeuromatchAcademy-exercise#1]
a convolutional neural network stacked onto a 'Subject Layer' and trained with a constractive objective to predict the deep representations of the audio waveform learnt by a dedicated module pretrained on 56k hours of speech. 

![f8ba39009024ad0fb9dac8c46bbf808](https://user-images.githubusercontent.com/72961328/216806745-84c69802-1a96-427a-b609-7aad0a6cde3e.png)

**Comment by Serendipityzzz** [comment]
C is channel/sensor, T is time, X $\in$ $\mathbb{R}^{C \times T}$ be a segment of a brain recording of a given subject while she listens to a speech segment of the same duraiton. Y $\in$ $\mathbb{R}^{F \times T}$ be the latent representation of speech, here the Mel spectrogram with F frequency bands. Thus supervised decoding consists of finding a decoding function: $f_{reg}: \mathbb{R}^{C \times T} \rightarrow \mathbb{R}^{F \times T}$ such that  $f_{reg}$ predicts Y given X. We denote $\hat{Y} = f_{reg}(X)$ the representation of speech from the brain, $f_{reg}$ belongs to models like DNN, then a regression loss looks like
![c78f9c81441ef817f4b1a1af3845d7e](https://user-images.githubusercontent.com/72961328/216814664-ccba6c3a-2641-451c-a851-f8332903911c.png)

**Comment by Serendipityzzz** [comment]
But this regression loss faces several challenges: decoding predictions appear to be dominated by a non-distinguishable broadband component when speech is present. So Meta AI made three main contributions: the introduction of a contrastive loss, a pre-trained deep speech representation, and a dedicated brain decoder.

**Comment by Serendipityzzz** [comment]
if we want to change the loss function, we can use NCE
https://zhuanlan.zhihu.com/p/334772391
